### PR TITLE
Migrate utils out of ethservice

### PR DESCRIPTION
**Pull request by blurpesec**
### Description

Migrates utils from the ethservice folder, removes unneeded logic, and adds more test coverage

### CODE REVIEW

File: src/utils/normalise.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import uts46 from 'idna-uts46';
+
+export const normalise = (name: string): string =>
```
Comment (FrederikBolding) [inline src/utils/normalise.ts:3]: Please use https://github.com/MyCryptoHQ/MyCrypto/blob/0e0b2610ff994fdac59e9885eec7189f8c3b02a3/src/services/EthService/utils/normalize.ts and perhaps move that out. We don't want the duplication.

File: src/components/GeneralLookupField.tsx [hunk 2]
```diff
@@ -3,7 +3,8 @@ import React, { useCallback, useEffect, useRef, useState } from 'react';
 import { ResolutionError } from '@unstoppabledomains/resolution/build/resolutionError';
 
 import { DomainStatus, InlineMessage } from '@components';
-import { isValidENSName, isValidETHAddress } from '@services/EthService';
+import { isValidETHAddress } from '@services/EthService';
+import { isValidENSName } from '@services/EthService/ens';
```
Comment (FrederikBolding) [inline src/components/GeneralLookupField.tsx:7]: Why change this if it hasn't been moved out?

**Comment by FrederikBolding** [comment]
Rebase please so we can see the actual coverage

File: spec/libs/ens.spec.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
-import { normalize } from '@services';
 import { getNameHash } from '@services/EthService/ens';
+import { normalize } from '@services/EthService/utils/normalize';
```
Comment (FrederikBolding) [inline spec/libs/ens.spec.ts:2]: Move to utils

File: src/components/TransactionFlow/ConfirmTransaction.tsx [hunk 4]
```diff
@@ -317,23 +317,23 @@ export const ConfirmTransactionUI = ({
...
-            </>
-          )}
+              <>
+                <AssetIcon uuid={asset.uuid} size={'25px'} />
+                <Amount
+                  assetValue={`${amount} ${asset.ticker}`}
+                  bold={true}
+                  baseAssetValue={`+ ${totalEtherEgress} ${baseAsset.ticker}`}
+                  fiat={{
+                    symbol: getFiat(settings).symbol,
```
Comment (FrederikBolding) [inline src/components/TransactionFlow/ConfirmTransaction.tsx:327]: ?

File: src/components/TransactionFlow/displays/TransactionDetailsDisplay.tsx [hunk 5]
```diff
@@ -182,10 +186,10 @@ function TransactionDetailsDisplay({
                   <CopyableCodeBlock>{data}</CopyableCodeBlock>
                 </div>
               ) : (
-                <div className="TransactionDetails-row-data-empty">
-                  {translate('TRANS_DATA_NONE')}
-                </div>
-              )}
+                  <div className="TransactionDetails-row-data-empty">
+                    {translate('TRANS_DATA_NONE')}
```
Comment (FrederikBolding) [inline src/components/TransactionFlow/displays/TransactionDetailsDisplay.tsx:190]: ?

File: src/features/InteractWithContracts/components/Interact.tsx [hunk 6]
```diff
@@ -13,7 +13,7 @@ import {
   NetworkSelectDropdown
 } from '@components';
 import { getNetworkById, useNetworks } from '@services';
-import { isValidENSName } from '@services/EthService';
+import { isValidENSName } from '@services/EthService/ens';
```
Comment (FrederikBolding) [inline src/features/InteractWithContracts/components/Interact.tsx:16]: ?

File: src/services/EthService/ens/index.ts [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import { keccak } from 'ethereumjs-util';
 
-import { normalize } from '../utils';
+import { normalize } from '../utils/normalize';
```
Comment (FrederikBolding) [inline src/services/EthService/ens/index.ts:3]: Related to other normalize issues

File: src/services/EthService/ens/validators.ts [hunk 8]
```diff
@@ -1,4 +1,4 @@
-import { normalize } from '@services/EthService';
+import { normalize } from '../utils/normalize';
```
Comment (FrederikBolding) [inline src/services/EthService/ens/validators.ts:1]: Related

File: src/services/WalletService/deterministic/mnemonic.ts [hunk 9]
```diff
@@ -1,6 +1,6 @@
 import { default as Wallet } from 'ethereumjs-wallet';
 
-import { decryptMnemonicToPrivKey } from '@services/EthService';
```
Comment (FrederikBolding) [inline src/services/WalletService/deterministic/mnemonic.ts:3]: Can you explain by which criteria you have extracted to `@utils`? This seems like something the `EthService` should do?

File: src/utils/transactions.ts [hunk 10]
```diff
@@ -2,16 +2,18 @@ import { addHexPrefix } from 'ethereumjs-util';
 import { Optional } from 'utility-types';
```
Comment (FrederikBolding) [inline src/utils/transactions.ts:2]: Maybe merge `transactions.ts` and `transaction.ts`?

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

File: src/components/TransactionFlow/ConfirmTransaction.tsx (hunk 4 above)
Comment (blurpesec) [reply src/components/TransactionFlow/ConfirmTransaction.tsx:327]: linter :shrug:

File: src/components/TransactionFlow/displays/TransactionDetailsDisplay.tsx (hunk 5 above)
Comment (blurpesec) [reply src/components/TransactionFlow/displays/TransactionDetailsDisplay.tsx:190]: also linter :woman_shrugging:

File: src/services/WalletService/deterministic/mnemonic.ts (hunk 9 above)
Comment (blurpesec) [reply src/services/WalletService/deterministic/mnemonic.ts:3]: Extracted out shared utils (used in multiple places), and utils that aren't used in the EthService directory.

File: src/services/EthService/ens/validators.ts [hunk 11]
```diff
@@ -1,4 +1,4 @@
-import { normalize } from '@services/EthService';
+import { normalize } from '@utils/normalize';
```
Comment (FrederikBolding) [inline src/services/EthService/ens/validators.ts:1]: Is it not exported?

File: src/utils/transaction.ts [hunk 12]
```diff
@@ -350,3 +366,69 @@ export const deriveTxRecipientsAndAmount = (
   }
   return { to: toAddress, amount: value, receiverAddress: toAddress };
 };
+
+export const appendSender = (senderAddress: ITxFromAddress) => async (
```
Comment (FrederikBolding) [inline src/utils/transaction.ts:370]: Can we test these functions?

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

File: src/utils/transaction.ts (hunk 12 above)
Comment (blurpesec) [reply src/utils/transaction.ts]: le done.

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**